driver-async/src/examples/gridfs/GridFSTour.java

```diff
@@ -0,0 +1,303 @@
+        final CountDownLatch uploadLatch = new CountDownLatch(1);
+        gridFSBucket.uploadFromStream("mongodb-tutorial", streamToUploadFrom, options, new SingleResultCallback<ObjectId>() {
+            @Override
+            public void onResult(final ObjectId result, final Throwable t) {
+                fileIdRef.set(result);
+                System.out.println("The fileId of the uploaded file is: " + result.toHexString());
+                streamToUploadFrom.close(new SingleResultCallback<Void>() {
+                    @Override
+                    public void onResult(final Void result, final Throwable t) {
+                        uploadLatch.countDown();
+                    }
+                });
+            }
+        });
+        uploadLatch.await();
+
+        ObjectId fileId = fileIdRef.get();
+
+        /*
+         * OpenUploadStream Example
+         */
+
+        // Get some data to write
+        ByteBuffer data = ByteBuffer.wrap("Data to upload into GridFS".getBytes(StandardCharsets.UTF_8));
+
+        final CountDownLatch uploadLatch2 = new CountDownLatch(2);
+        final GridFSUploadStream uploadStream = gridFSBucket.openUploadStream("sampleData");
+        uploadStream.write(data, new SingleResultCallback<Integer>() {
+            @Override
+            public void onResult(final Integer result, final Throwable t) {
+                uploadLatch2.countDown();
+                System.out.println("The fileId of the uploaded file is: " + uploadStream.getFileId().toHexString());
+
+                uploadStream.close(new SingleResultCallback<Void>() {
+                    @Override
+                    public void onResult(final Void result, final Throwable t) {
+                        uploadLatch2.countDown();
+                    }
+                });
+            }
+        });
+
+        uploadLatch2.await();
+
+        /*
+         * Find documents
+         */
+        final CountDownLatch findLatch = new CountDownLatch(1);
+        System.out.println("File names:");
+        gridFSBucket.find().forEach(new Block<GridFSFile>() {
+            @Override
+            public void apply(final GridFSFile gridFSFile) {
+                System.out.println(" - " + gridFSFile.getFilename());
+            }
+        }, new SingleResultCallback<Void>() {
+            @Override
+            public void onResult(final Void result, final Throwable t) {
+                findLatch.countDown();
+            }
+        });
+        findLatch.await();
+
+        /*
+         * Find documents with a filter
+         */
+        final CountDownLatch findLatch2 = new CountDownLatch(1);
+        gridFSBucket.find(eq("metadata.contentType", "image/png")).forEach(
+                new Block<GridFSFile>() {
+                    @Override
+                    public void apply(final GridFSFile gridFSFile) {
+                        System.out.println(gridFSFile.getFilename());
```

Comment:
I'd rather not add lots of code to add extra files to gridfs because there is no match in the walkthrough - its just an example of using the find iterable and is the same as the sync version.